models/review.go

```diff
@@ -442,3 +446,117 @@ func InsertReviews(reviews []*Review) error {
 
 	return sess.Commit()
 }
+
+// AddRewiewRequest add a review request from one reviewer
+func AddRewiewRequest(issue *Issue, reviewer *User, doer *User) (comment *Comment, err error) {
+	reviews, err := GetReviewersByIssueID(issue.ID)
+	if err != nil {
+		return
+	}
+
+	var get *Review
+	for _, rev := range reviews {
+		if rev.ReviewerID == reviewer.ID {
+			get = rev
+			break
+		}
+	}
+
+	// skip it when reviewer hase been request to review
+	if get != nil && get.Type == ReviewTypeRequest {
+		return nil, nil
+	}
+
+	sess := x.NewSession()
+	defer sess.Close()
+	if err := sess.Begin(); err != nil {
+		return nil, err
+	}
+
+	var official bool
+	official, err = isOfficialReviewer(sess, issue, reviewer)
+
+	if err != nil {
+		return nil, err
+	}
+
+	if !official {
+		official, err = isOfficialReviewer(sess, issue, doer)
```

Comment:
Interesting... this would make the request "official" if the doer is an official reviewer... I didn't see this coming, but it looks like the right thing to do. 👍